packages/async-subscription/src/stateful.ts

```diff
@@ -14,46 +9,26 @@ export function makeStatefulSubscribable<T>(
   // for unsubscribe callbacks received from subscription.subscribe().
   retain(subscription);
 
-  let current = subscription.initial;
-  let listening = true;
-
-  const subscribers = new Set<Subscriber<T>>();
-
-  const subscriptionResult = Promise.resolve<RemoteSubscribeResult<T>>(
-    subscription.subscribe(listener),
-  ).then((result) => {
-    listener(result[1]);
-    return result;
-  });
+  const current = subscription.initial;
 
   return {
     get current() {
       return current;
     },
     subscribe(subscriber) {
-      subscribers.add(subscriber);
-
-      return () => {
-        subscribers.delete(subscriber);
-      };
+      const subscriptionResult = Promise.resolve(
+        subscription.subscribe((value) => {
+          subscriber(value);
+        }),
+      ).then((result) => {
+        subscriber(result[1]);
```

Comment:
I think this refactor will cause some tests in checkout-web to fail because the first subscription will automatically call the callback, unlike the previous implementation